project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module

```diff
@@ -677,15 +683,168 @@ function c4m_user_access_denied() {
+ * Create 'Contact User' form.
+ */
+function c4m_user_contact_form($form, &$form_state, $account) {
+  global $user;
+  $form_state['addressee_user'] = $account;
+
+  $form['c4m_contact_user_from'] = array(
+    '#type' => 'textfield',
+    '#title' => t('From'),
+    '#description' => t('From address.'),
+    '#size' => 30,
+    '#default_value' => $user->mail,
+    '#disabled' => TRUE,
+  );
+
+  $form['c4m_contact_user_subject'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Subject'),
+    '#description' => t('Enter mail subject.'),
+    '#size' => 10,
+  );
+
+  $form['c4m_contact_user_body'] = array(
+    '#title' => t('Body'),
+    '#description' => t('Enter mail body.'),
+    '#type' => 'text_format',
+    '#format' => 'full_html',
+  );
+
+  $form['c4m_contact_user_self_copy'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Send me a copy'),
+    '#default_value' => FALSE,
   );
+
+  $form['c4m_contact_user_submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Send'),
+  );
+
+  return $form;
+}
+
+/**
+ * Validate function for contact_form.
+ *
+ * Verifies that email subject and body are not empty.
+ */
+function c4m_user_contact_form_validate($form, $form_state) {
+  $subject = $form_state['values']['c4m_contact_user_subject'];
+  $body = $form_state['values']['c4m_contact_user_body']['value'];
+
+  if (empty($subject)) {
+    form_error($form['c4m_contact_user_subject'], 'The subject of the mail can\'t be empty.');
+  }
+
+  if (empty($body)) {
+    form_error($form['c4m_contact_user_body'], 'The body of the mail can\'t be empty.');
+  }
+}
+
+/**
+ * Submit function for contact_form.
+ *
+ * Generates required parameters and sends an email.
+ * If 'Send copy' checkbox is checked, sends copy mail to running user.
+ */
+function c4m_user_contact_form_submit($form, $form_state) {
+  global $user;
+
+  // Generate and send mail.
+  $addressee_account = $form_state['addressee_user'];
+  $subject = $form_state['values']['c4m_contact_user_subject'];
+  $body = $form_state['values']['c4m_contact_user_body']['value'];
+
+  _c4m_user_send_mail($addressee_account->mail, $user->mail, $subject, $body);
+
+  if ($form_state['values']['c4m_contact_user_self_copy']) {
+    // Sending a copy of the mail to the sender since the user checked
+    // the "Send me a copy" checkbox on the form.
+    _c4m_user_send_mail($user->mail, $user->mail, "Self copy: {$subject}", $body);
+  }
+}
+
+/**
+ * Generates and sends an email.
+ *
+ * @param string $to
+ *   Email destination address.
+ * @param string $from
+ *   Email source address - current user email.
+ * @param string $subject
+ *   Email message subject.
+ * @param string $body
+ *   Email message body.
+ */
+function _c4m_user_send_mail($to, $from, $subject, $body) {
+  global $language;
+  $module = 'c4m_user';
+  $key = 'user_contact_mail';
+  $params = array();
+  $send = FALSE;
+
+  $message = drupal_mail($module, $key, $to, $language, $params, $from, $send);
```

Comment:
@anvmn 
A better solution will be to replace `$language` with `user_preferred_language($account)`